src/cargo/ops/cargo_read_manifest.rs

```diff
@@ -183,11 +183,15 @@ fn read_nested_packages(
             v.insert(pkg);
         }
         Entry::Occupied(_) => {
-            let _ = config.shell().warn(format!(
-                "skipping duplicate package `{}` found at `{}`",
-                pkg.name(),
-                path.to_string_lossy()
-            ));
+            if let Some(registries) = pkg.publish() {
+                if registries.len() == 0 {
```

Comment:
> Restrict duplicate deps warning only to unpublished packages 

I was thinking we'd want to do the opposite, to only show the warning for published packages.  What are your thoughts on one vs the other?

Reply:
Oh wait, I thought I was suppose to do this as per the last comment on the related issue. Sorry, I'll change it.
As for my opinion, I don't think I'm familiar enough with cargo and Rust ecosystem to have an opinion on this, so I'll just follow the cargo's team decision.

Reply:
Bah, got it backwards in that comment; sorry about that

Reply:
Speaking of, we should probably have a comment with our intention which I don't think was in the Issue.  We can assume a `publish = false` isn't intended to be seen by users so we can hide the warning about those since the user is unlikely to care about those cases.

Reply:
Ah I see, thank you for letting me know the reasoning behind it!